netbox/project-static/src/netbox.ts

```diff
@@ -42,7 +41,16 @@ function initWindow(): void {
   const documentForms = document.forms
   for (var documentForm of documentForms) {
     if (documentForm.method.toUpperCase() == 'GET') {
-      documentForm.addEventListener('submit', cleanGetUrl)
+// @ts-ignore: // formdata is not yet supported by TS https://github.com/microsoft/TypeScript/issues/36217
+      documentForm.addEventListener('formdata', function(event) {
+// @ts-ignore: // formdata is not yet supported by TS https://github.com/microsoft/TypeScript/issues/36217
+        let formData = event.formData;
+// @ts-ignore: // formdata is not yet supported by TS https://github.com/microsoft/TypeScript/issues/36217
+        for (let [name, value] of Array.from(formData.entries())) {
+// @ts-ignore: // formdata is not yet supported by TS https://github.com/microsoft/TypeScript/issues/36217
+          if (value === '') formData.delete(name);
+        }
+      });
```

Comment:
```suggestion
      // @ts-ignore: Our version of typescript seems to be too old for FormDataEvent
      documentForm.addEventListener('formdata', function (event: FormDataEvent) {
        let formData: FormData = event.formData;
        for (let [name, value] of Array.from(formData.entries())) {
          if (value === '') formData.delete(name);
        }
      });
```

Reply:
I spent a little time trying to get this working. The suggested change works on my PC.

FormDataEvent is actually in typescript, our version is just too old. I tried updating typescript but as usual you just enter a rabbithole of broken stuff. Updating typescript -> Now popperjs has type errors -> Try to update popperjs -> Didn't work -> Google popperjs -> It changed name to floating-ui -> Start reading migration guide -> Give up.

I think we need to spend a little time soon getting all our JS dependencies up-to-date. Our typescript release seems to be from june 2021, which might as well be a decade in web-framework terms.

Could you try my suggestion? See if CI builds it, and then maybe I'll take a look at the dependency situation soon-ish.

Reply:
My tsc throws an error : 
```
tsc --noEmit
src/netbox.ts(46,30): error TS2339: Property 'formData' does not exist on type 'Event'.
src/netbox.ts(47,18): error TS2488: Type 'unknown' must have a '[Symbol.iterator]()' method that returns an iterator.
```
the ts-ignore is only for the next lines... so it also fails for the other items it does not know

Reply:
If you push the bundles so that the CI doesn't fail and it builds I think it's fine for now. I'll fix it up when I get to updating our JS dependencies.

I'm a little worried that my suggesting didn't work for you, so please remove your node_modules folder and do a clean `yarn install` afterwards to make sure you have the correct dependencies. Do this before bundling.